oxidation/libparsec/crates/client_types/src/path.rs

```diff
@@ -0,0 +1,69 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use serde::Serialize;
+use std::{
+    fmt::Debug,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize)]
+pub struct StrPath(PathBuf);
+
+impl std::str::FromStr for StrPath {
+    // Type Err require display for generator
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(s.into())
+    }
+}
+
+impl AsRef<str> for StrPath {
```

Comment:
This is against the naming convention.
https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

As taking the reference here is not free but expensive

Reply:
Yes I know, but the generator need this trait for Path

Reply:
Could this implementation be lock only on `wasm32` ?

Reply:
Done